nessie/jobs/sync_canvas_requests_snapshots.py

```diff
@@ -0,0 +1,111 @@
+"""
+Copyright ©2019. The Regents of the University of California (Regents). All Rights Reserved.
+
+Permission to use, copy, modify, and distribute this software and its documentation
+for educational, research, and not-for-profit purposes, without fee and without a
+signed licensing agreement, is hereby granted, provided that the above copyright
+notice, this paragraph and the following two paragraphs appear in all copies,
+modifications, and distributions.
+
+Contact The Office of Technology Licensing, UC Berkeley, 2150 Shattuck Avenue,
+Suite 510, Berkeley, CA 94720-1620, (510) 643-7201, otl@berkeley.edu,
+http://ipira.berkeley.edu/industry-info for commercial licensing opportunities.
+
+IN NO EVENT SHALL REGENTS BE LIABLE TO ANY PARTY FOR DIRECT, INDIRECT, SPECIAL,
+INCIDENTAL, OR CONSEQUENTIAL DAMAGES, INCLUDING LOST PROFITS, ARISING OUT OF
+THE USE OF THIS SOFTWARE AND ITS DOCUMENTATION, EVEN IF REGENTS HAS BEEN ADVISED
+OF THE POSSIBILITY OF SUCH DAMAGE.
+
+REGENTS SPECIFICALLY DISCLAIMS ANY WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE. THE
+SOFTWARE AND ACCOMPANYING DOCUMENTATION, IF ANY, PROVIDED HEREUNDER IS PROVIDED
+"AS IS". REGENTS HAS NO OBLIGATION TO PROVIDE MAINTENANCE, SUPPORT, UPDATES,
+ENHANCEMENTS, OR MODIFICATIONS.
+"""
+
+import time
+
+from flask import current_app as app
+from nessie.externals import canvas_data, s3
+from nessie.jobs.background_job import BackgroundJob, BackgroundJobError
+from nessie.lib import metadata
+from nessie.lib.dispatcher import dispatch
+
+"""Logic for Canvas snapshot sync job."""
+
+
+def delete_objects_with_prefix(prefix, whitelist=[]):
+    keys_to_delete = []
+    existing_keys = s3.get_keys_with_prefix(prefix)
+    if existing_keys is None:
+        app.logger.error('Error listing keys, aborting job.')
+        return False
+    for key in existing_keys:
+        filename = key.split('/')[-1]
+        if filename not in whitelist:
+            keys_to_delete.append(key)
+    app.logger.info(
+        f'Found {len(existing_keys)} key(s) matching prefix "{prefix}", {len(existing_keys) - len(keys_to_delete)} '
+        f'key(s) in whitelist, will delete {len(keys_to_delete)} object(s)')
+    if not keys_to_delete:
+        return True
+    if s3.delete_objects(keys_to_delete):
+        metadata.delete_canvas_snapshots(keys_to_delete)
+        return True
+    else:
+        return False
```

Comment:
@sandeepmjay It looks like this method is the same as the identically named method in the `SyncCanvasSnapshots` job. If this job in fact needs this method, I'd suggest moving it into a shared location.

Reply:
makes sense moving it to a shared s3_utils location.